===
@try, catch, finally
===
@try
{
    throw new InvalidOperationException("You did something invalid.");
}
catch (Exception ex)
{
    <p>The exception message: @ex.Message</p>
}
finally
{
    <p>The finally statement.</p>
}
---

(compilation_unit
  (razor_try
    (throw_statement
      (object_creation_expression
        (identifier)
        (argument_list
          (argument
            (string_literal
              (string_literal_content))))))
    (razor_catch
      (catch_declaration
        (identifier)
        (identifier))
      (element
        (razor_implicit_expression
          (member_access_expression
            (identifier)
            (identifier)))))
    (razor_finally
      (element))))

===
@try, catch
===
@try
{
  ThisCan.Fail();
}
catch (Exception)
{
    <p>Operation Failed</p>
}
---

(compilation_unit
  (razor_try
    (expression_statement
      (invocation_expression
        (member_access_expression
          (identifier)
          (identifier))
        (argument_list)))
    (razor_catch
      (catch_declaration
        (identifier))
      (element))))
